schema/cat-vrs/profiles/cat-vrs-profiles-source.yaml

```diff
@@ -0,0 +1,68 @@
+$schema: "https://json-schema.org/draft/2020-12/schema"
+$id: "https://w3id.org/ga4gh/schema/cat-vrs/1.x/profiles/cat-vrs-profiles-source.yaml"
+title: GA4GH-Cat-VRS-Definitions
+type: object
+
+imports:
+  gks.core-im: ../../core-im/core-im-source.yaml
+  gks.data-types: ../../data-types/data-types-source.yaml
+  vrs: ../../vrs/vrs-source.yaml
+  cat.core: ../core/cat-vrs-core-source.yaml
+
+$defs:
+  ProteinSequenceConsequenceProperties:
+    type: object
+    maturity: draft
+    description: >-
+      Categorical Variant constraints found in Protein Sequence Consequences.
+    properties:
+      constraints:
+        contains:
+          allOf:
+            - $ref: "/ga4gh/schema/cat-vrs/1.x/core/json/DefiningContextConstraint"
+            - type: object
+              properties:
+                relations:
+                  contains:
+                    const: codon_translation
+
+  ProteinSequenceConsequence:
+    description: >-
+      A change that occurs in a protein sequence as a result of genomic changes. Due to the degenerate nature
+      of the genetic code, there are often several genomic changes that can cause a protein sequence consequence.
+      The protein sequence consequence, like a :ref:`CanonicalAllele`, is defined by an
+      `Allele <https://vrs.ga4gh.org/en/2.x/concepts/MolecularVariation/Allele.html#>`_
+      that is representative of a collection of congruent Protein Alleles that share the same altered codon(s).
+    allOf:
+      - $ref: "#/$defs/ProteinSequenceConsequenceProperties"
+      - $ref: "/ga4gh/schema/cat-vrs/1.x/core/json/CategoricalVariant"
+
+  CanonicalAlleleProperties:
+    type: object
+    maturity: draft
+    description: >-
+      Categorical Variant constraints found in Canonical Alleles.
+    properties:
+      constraints:
+        contains:
+          allOf:
+            - $ref: "/ga4gh/schema/cat-vrs/1.x/core/json/DefiningContextConstraint"
+            - type: object
+              properties:
+                relations:
+                  allOf:
+                    - contains:
+                        const: sequence_liftover
+                    - contains:
+                        const: transcript_projection
+
+  CanonicalAllele:
```

Comment:
@ahwagner (FIRST) This rework is incredible - thank you. It's a big effort. (SECOND) I think these CatVar profiles should be 'type'd with the subtype of CatVar that they are. If not, we should probably put a subtype field on them maybe?  I think bundling all `CategoricalVariant`s under one `type: CategoricalVariant` enumerated property diminishes the value of the `type` field.  Even though the structure is no different for any subclass, the `constraints` are. And it's the set of `constraints` that define a new type of `CategoricalVariant`.  Just my $.02.

Reply:
we can move this to discussions.

Reply:
I'm happy to discuss this Larry!  I actually need to stew a bit and get my throughts straight on what type profiles should be as well.  Because they describe `CategoricalVariant` objects, but the profile itself is a effectively a function from `CategoricalVariants` to boolean truth values (yes, this CatVAr is a member of this profile, vs no, it is not).

I see we've split out the profiles from the `CategoricalVariant` and constraint classes.  That makes sense to me, for similar reasons as my thoughts above.